File: buff/classes/472/pub/ch1/intf/TryTable.java
// javac TryTable.java && java TryTable

public class TryTable {

    public static void main(String[] args) {
	Table<String,Integer> t=new HashTable<String,Integer>();
	t.put("Hi",100);
	System.out.println(t.get("Hi"));
    }

}

File: buff/classes/472/pub/ch1/intf/TryTable.cc
// g++ -Wall -g -o TryTable TryTable.cc && ./TryTable

#include <iostream>

using namespace std;

#include "HashTable.h"

int main() {
  // program to an interface!
  // NOT: auto t=...
  Table<string,int>* t=new HashTable<string,int>();
  string k="Hi";
  int d=100;
  t->put(k,d);
  cout << t->get(k) << endl;
  delete t;
  return 0;
}
